server/api/utils/clients/iguazio.py

```diff
@@ -191,50 +278,43 @@ def delete_project(
         deletion_strategy: mlrun.common.schemas.DeletionStrategy = mlrun.common.schemas.DeletionStrategy.default(),
         wait_for_completion: bool = True,
     ) -> bool:
-        self._logger.debug(
-            "Deleting project in Iguazio",
-            name=name,
-            deletion_strategy=deletion_strategy,
-        )
-        body = self._transform_mlrun_project_to_iguazio_project(
-            mlrun.common.schemas.Project(
-                metadata=mlrun.common.schemas.ProjectMetadata(name=name)
+        with self._job_cache.get_delete_lock(name):
+            # check if project is already being deleted
+            job_id = self._job_cache.get_delete_job_id(name)
+
+            # the existing job might have already been completed, but the cache was not yet invalidated
+            # in that case, we want to create a new deletion job
+            if job_id and self._is_job_done(session, job_id):
+                self._job_cache.remove_delete_job(name)
```

Comment:
you need to acquire the lock and ensure you are deleting the correct job id

Reply:
The lock is already acquired in this block, a few lines above it :
```
with self._job_cache.get_delete_lock(name):
```